src/components/MessageList/utils.ts

```diff
@@ -71,18 +73,23 @@ export const processMessages = <
     if (
       i > 0 &&
       !disableDateSeparator &&
-      !threadList &&
       previousMessage.created_at &&
-      isDate(previousMessage.created_at)
+      isDate(previousMessage.created_at) &&
+      (!threadList || (threadList && enableThreadDateSeparator))
```

Comment:
These conditions get cryptic and hard to understand at times - I wouldn't mind in-code eplanation comments - what do you guys think? @MartinCupela @petyosi

Reply:
Agreed. @arnautov-anton feel ree to add a suggestion. I usually create extra variables with descriptive names as means to document intent.

Reply:
@arnautov-anton @petyosi I have added comments. Not just to explain the conditions, but also to document the `processMessages` function and its params + tests.